### PR TITLE
Ruby Milestone -2 : Ruby Enumerables

### PR DESCRIPTION
**Pull request by balerum03**
On this project, we had to create a custom Enumerable method using Ruby including:
- #my_each
- #my_each_with_index
- #my_select
- #my_all?
- #my_any?
- #my_none?
- #my_count
- #my_map
- #my_inject
- #multiply_els

### CODE REVIEW

**Review by davidmrtz-dev — CHANGES_REQUESTED** [review]
## Hi Team.
### Very good job so far.

However, there are some issues with the my_any and my_inject methods that need to be fixed, please check out below.  👨‍💻  ⬇️ 

****

### **Please make the changes required and submit for re-review once you are ready!** 

****

### Status: Changes required. ♻️  
****

Happy coding!  💻  


**[TSE:  David Elí]** 
Slack: @ David Eli Martinez Garcia

**Comment by Berabjesus** [comment]
@davitomix Thank you for the feedback but I can't see what the requested change is.

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      my_each { |i| return false unless i.is_a?(args) }
+    elsif args.nil? && block_given?
+      my_each { |i| return false unless yield i }
+    else
+      my_each { |i| return false if i.nil? || i == false }
+    end
+    true
+  end
+
+  def my_any?(args = nil)
```
Comment (davidmrtz-dev) [inline main.rb:44]: my_any? method should return false if at least one of the collection is not true.
[Reference](https://ruby-doc.org/core-2.6.5/Enumerable.html).
Screen Shot 2020-08-04 at 7 08 13 PM
****
my_any? method when Regex is given as an argument it should returns true if any of the collection matches the Regex.
Screen Shot 2020-08-04 at 7 12 34 PM

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    new_array = []
+    if args.nil?
+      my_each_with_index { |i, j| new_array[j] = yield i }
+    else
+      my_each_with_index { |i, j| new_array[j] = args.call(i) }
+    end
+    new_array
+  end
+
+  def my_inject(arg1 = nil, arg2 = nil)
```
Comment (davidmrtz-dev) [inline main.rb:104]: my_inject methods should raise a "LocalJumpError" when no block or argument id is given, we got "Argument error" instead.
Screen Shot 2020-08-04 at 7 18 16 PM

File: main.rb (hunk 1 above)
Comment (Berabjesus) [reply main.rb:44]: ## @davitomix thank you for the feedback. The requested change has been made.

File: main.rb (hunk 2 above)
Comment (Berabjesus) [reply main.rb:104]: ## @davitomix thank you for the feedback. The requested change has been made.

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Thank you for applying all the suggested changes :smirk_cat: 
You can merge it now :+1: and move forward
Good continuation :handshake: 

Thanks
Mael FOSSO

**Pull request merged**